### ISSUE TITLE
skip test that uses vcr if casssette is not UTF-8

### ISSUE THREAD

**Issue by sckott** [ropensci/vcr#99]
Arose from recent realization that strings in cassettes, any string in the cassette (not just the response body), that are not UTF-8 can cause problems on some platforms, e.g., 

- rcrossref (with commit 7ac23d58eca2e3a125e6dc0879d1350730097041)
    - `tools::showNonASCIIfile("tests/fixtures/cr_citation_count.yml")`
    - `tools::showNonASCIIfile("tests/fixtures/cr_cn_different_base_urls.yml")`
- rbison (with commit ba1241005fce12f555f3902bbecfcca3dec8294c)
    - `tools::showNonASCIIfile("tests/fixtures//bison.yml")`
- rdatacite (with commit 2f409f9c56a0a6eba5ac53246c7bce153c6fe552)
    - `tools::showNonASCIIfile("tests/fixtures/vcr_cassettes/dc_search_csv.yml")`

----

Options:

- include helper fxn exported from this pkg to skip tests with vcr usage, e.g., `skip_vcr_non_utf8()`
- no fxn, but document the issue and examples of how to fix
- vcr exported fxn that behaves similar to the privacy key filtering tool to fix utf-8 strings in cassettes prior to reading them if possible? or something like that

**Comment by sckott** [comment]
I think best approach is to take the string from the cassette and replace any non ascii strings with unicode equivalent, or if none found, then drop the non-ascii, or something else?  ideally would use `stringi`, but the pkg is so heavy so not sure want to import it. but could have as a Suggest and if not installed, then do something with base R. I like the idea of a filter before loading into yaml because then we don't touch the real request/response

probably happen on the line after this or so https://github.com/ropensci/vcr/blob/master/R/serializers-yaml.R#L76 

maybe: 

```r
text_cleaner <- function(x) {
    if (requireNamespace("stringi")) {
        stringi::stri_escape_unicode(x)
    } else {
        # some other w/o stringi
    }
}
```

**Comment by sckott** [comment]
another reason to get that JSON serializer done sooner than later, it may help avoid these yaml pkg problems

**Comment by maelle** [comment]
> another reason to get that JSON serializer done sooner than later, it may help avoid these yaml pkg problems

It seems so at least for my encoding problems. :-D

**Comment by sckott** [comment]
right. it'd be nice to have a better yaml pkg, but its easier here to just recommend people use json if they have any issue
